maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/PluginsMetadataGenerator.java

```diff
@@ -135,14 +136,34 @@ private PluginInfo extractPluginInfo(Artifact artifact) {
                             String artifactId = root.getChild("artifactId").getValue();
                             String goalPrefix = root.getChild("goalPrefix").getValue();
                             String name = root.getChild("name").getValue();
-                            return new PluginInfo(groupId, artifactId, goalPrefix, name);
+                            // sanity check: plugin descriptor extracted from artifact must have same GA
+                            if (Objects.equals(artifact.getGroupId(), groupId)
+                                    && Objects.equals(artifact.getArtifactId(), artifactId)) {
+                                return new PluginInfo(groupId, artifactId, goalPrefix, name);
+                            } else {
+                                throw new InvalidArtifactPluginMetadataException(
+                                        "Artifact " + artifact.getGroupId() + ":"
+                                                + artifact.getArtifactId()
+                                                + " JAR (to be installed/deployed) contains Maven Plugin metadata for plugin "
+                                                + groupId + ":" + artifactId + "; coordinates are conflicting. "
+                                                + "Most probably your JAR contains rogue Maven Plugin metadata, "
+                                                + "possible causes may be: shaded in Maven Plugin or some rogue resource)");
+                            }
                         }
                     }
+                } catch (RuntimeException e) {
```

Comment:
Which for example?

Reply:
IAPMEx? But lets have NPE and other fail the build as well, as below it is in comment, IOEx, PlexusXMLEx etc are and were swallowed.

Reply:
But why catch and rethrow?

Reply:
As other catch below catches all?

Reply:
I might be missing something, intent is "throw all unexpected ones" (or, swallow all expected ones, like IOEx, XML parsing and Plexus XML ones, these are all checked ones).

Also, the newly introduced IAPMEx is "unexpected" as well. All "unexpected" ones fails the build, while all the "expected" ones are swalled as were since 1.9.0.

Reply:
Then I would just rethrow `InvalidArtifactPluginMetadataException`.

Reply:
K, I will merge this PR as is, and later we can refine.